quantum/rgb_matrix/animations/pixel_flow_anim.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 @filterpaper
+// SPDX-License-Identifier: GPL-2.0+
+
+#ifdef ENABLE_RGB_MATRIX_PIXEL_FLOW
+RGB_MATRIX_EFFECT(PIXEL_FLOW)
+#    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS
+
+static bool PIXEL_FLOW(effect_params_t* params) {
+    // LED state array
+    static RGB led[DRIVER_LED_TOTAL];
+
+    static uint32_t wait_timer = 0;
+    if (wait_timer > g_rgb_timer) {
+        return false;
+    }
+
+    inline uint32_t interval(void) {
+        return 3000 / scale16by8(qadd8(rgb_matrix_config.speed, 16), 16);
+    }
+
+    if (params->init) {
+        // Clear LEDs and fill the state array
+        rgb_matrix_set_color_all(0, 0, 0);
+        for (uint8_t j = 0; j < DRIVER_LED_TOTAL; ++j) {
+            led[j] = (random8() & 2) ? (RGB){0,0,0} : hsv_to_rgb((HSV){random8(), qadd8(random8() >> 1, 127), rgb_matrix_config.hsv.v});
+        }
+    }
+
+    RGB_MATRIX_USE_LIMITS(led_min, led_max);
+    // Light LEDs based on state array
+    for (uint8_t i = led_min; i < led_max; ++i) {
+        RGB_MATRIX_TEST_LED_FLAGS();
+        rgb_matrix_set_color(i, led[i].r, led[i].g, led[i].b);
+    }
+
+    if (led_max == DRIVER_LED_TOTAL) {
```

Comment:
This comparison won't work properly in the `RGB_MATRIX_SPLIT` case (see how `RGB_MATRIX_USE_LIMITS()` is implemented — it always clamps `max` on the left size, so it won't reach `DRIVER_LED_TOTAL` there).

One option is to check for `params->iter == 0` instead (but then you will need to move thus code before the actual LED update, because it will run before the first iteration instead of after the last one). Another option is to use `!rgb_matrix_check_finished_leds(led_max)` (this test should work properly even with `RGB_MATRIX_SPLIT`).

However, even after those fixes you probably won't get the ideal behavior of this effect on a split board (both halves will have their own separate flows instead of one half flowing into another); but that might just look good enough to be still useful.